test/built-ins/AsyncFromSyncIteratorPrototype/throw/throw-null.js

```diff
@@ -51,7 +52,9 @@ asyncIterator.next().then(function() {
 }).then(function(result) {
   throw new Test262Error("Promise should be rejected, got: " + result.value);
 }, function(err) {
-  assert.sameValue(err, thrownError);
+  assert.sameValue(err.constructor, TypeError, "TypeError");
+  assert.sameValue(err instanceof TypeError, true);
+
   return asyncIterator.next().then(function(result) {
     assert.sameValue(result.value, undefined);
     assert.sameValue(result.done, true);
```

Comment:
Optional, but if you like - it might be a good opportunity to refactor these tests to use the [Async Helpers](https://github.com/tc39/test262/blob/main/rfcs/async-helpers.md). However, it's fine with me to land this PR without doing that.

As an example, I think this test would become something like
```js
asyncTest(async function () {
  await assert.throwsAsync(TypeError, () => asyncIterator.next(), "Promise should be rejected");
  const result = await asyncIterator.next();
  assert(result.done, "the iterator is completed");
  assert.sameValue(result.value, undefined, "value is undefined");
});
```
...which seems to me easier to follow.

Reply:
The latest commit refactors the tests to use Async Helpers as suggested.